### DISCUSSION TITLE
How to collapse nui.line(s) with their node?

### DISCUSSION THREAD

**Discussion by Otterpatsch** [MunifTanjim/nui.nvim#336] [Q&A] [answered]
Hello there,

Me again. So now i kinda got the formatting like i wanna have it and can also collapse my nodes via mapping (thats very great btw thank you for those features), but if i collapse a node only the child nodes are collapsed but not the lines of the node itself which is collapsed. 
How do i collapse the lines of a node but the very first one? Is this possible? 

My code for reference (https://github.com/Otterpatsch/bitbucket.nvim/blob/feature/add-discussion-section/lua/bitbucket/init.lua#L51) 
```
	local tree = Tree({
		bufnr = comment_split.bufnr,
		get_node_id = function(node)
			-- this is telling NuiTree where we're storing the id
			return node.id
		end,
		prepare_node = function(node)
			local line = Line()
			line:append(string.rep(" ", node:get_depth() - 1))
			if node:has_children() then
				line:append(node:is_expanded() and " " or " ", "SpecialChar")
			else
				line:append("-", "SpecialChar")
			end
			line:append(node.author .. " on " .. node.date)
			local lines = { line }

			for _, raw_line in ipairs(vim.split(node.text, "\n")) do
				table.insert(lines, Line({ Text(raw_line) }))
			end
			if not node:has_children() then
				table.insert(lines, Line({}))
			end
			return lines
		end,
	})
```

To collapse the current node (from the wiki):
```
	local map_options = { noremap = true, nowait = true }
	--- collpase current node ---
	comment_split:map("n", "g", function()
		local node = tree:get_node()

		if node:collapse() then
			tree:render()
		end
	end, map_options)
```
I thought okay lets just collapse .text from node but this is not working. So i would like to collapse all the lines but the very first one in lines. So how can i anyway collapse all the lines and then maybe just expand the first one (index 1 of lines)?

**Comment by MunifTanjim** [comment, marked as answer]
Maybe do this?

```lua
if node:is_expanded() then
  for _, raw_line in ipairs(vim.split(node.text, "\n")) do
    table.insert(lines, Line({ Text(raw_line) }))
  end
end
```

**Reply by Otterpatsch** [reply]
Exactly what i was looking for. Thank you!
